File: app/src/main/java/nl/acidcats/tumblrlikes/ui/screens/setup_screen/SetupScreenContract.java
package nl.acidcats.tumblrlikes.ui.screens.setup_screen;

import nl.acidcats.tumblrlikes.ui.screens.base.BasePresenter;
import nl.acidcats.tumblrlikes.ui.screens.base.BaseView;

/**
 * Created on 18/10/2018.
 */
public interface SetupScreenContract {
    String BLOG_EXT = ".tumblr.com";

    interface Presenter extends BasePresenter<View> {
        void onViewCreated();

        void checkCache();

        void onBlogTextChanged(String blog);

        void onSetupDone(String blog);

        void exportPhotos(String filename);
    }

    interface View extends BaseView {
        void setTumblrBlogText(String tumblrBlog);

        void enableCacheCheckButton(boolean enable);

        void enableOkButton(boolean enable);

        void showCacheMissToast(int cacheMissCount);

        void enableExportButton(boolean enable);

        void showExportCompleteToast(boolean success);
    }
}

File: app/src/main/java/nl/acidcats/tumblrlikes/ui/screens/setup_screen/SetupScreenContract.kt
package nl.acidcats.tumblrlikes.ui.screens.setup_screen

import nl.acidcats.tumblrlikes.ui.screens.base.BasePresenter
import nl.acidcats.tumblrlikes.ui.screens.base.BaseView

/**
 * Created on 18/10/2018.
 */
interface SetupScreenContract {

    interface Presenter : BasePresenter<View> {
        fun onViewCreated()

        fun onBlogTextChanged(blog: String)

        fun onSetupDone(blog: String)
    }

    interface View : BaseView {
        fun setTumblrBlogText(tumblrBlog: String)

        fun enableOkButton(enable: Boolean)
    }

    companion object {
        const val BLOG_EXT = ".tumblr.com"
    }
}
